### PR TITLE
Change: Pre-Commit and Pre-Push Hooks

### PR DESCRIPTION
**Pull request by martinmckenna**
## Description

* Updates pre-commit hooks
   * linting
   * fomatting
   * git add (the changes from the above commands)
   * circular dependency check
   * typechecking (on _all_ files, including non-staged ones) (*IMPORTANT:* [see here for why we can't add this to lint-staged](https://github.com/okonet/lint-staged/issues/468))
* Updates pre-push hooks
   * Runs _all_ unit tests
* Hoists Dependencies
  * TypeScript
  * tslint-configs (prettier and react)
* Creates 1 `tslint.json` as a single-source of truth.
   * find this in the root of the project

## Type of Change
- Breaking change ('break', 'deprecate')

## To Test

Run `yarn clean && yarn install:all` before committing new things

Make changes to some files. Try adding things the linter and typechecker will complain about (e.g. unsused variables or console logs) and try committing.

### CODE REVIEW

File: packages/manager/src/components/AbuseTicketBanner/AbuseTicketBanner.tsx [hunk 1]
```diff
@@ -47,13 +47,13 @@ export class AbuseTicketBanner extends React.Component<Props> {
...
 });
 
-const connected = connect(
+const connected = connect<ReduxStateProps, any, {}>(
   mapStateToProps,
   undefined
 );
 
-export default connected(AbuseTicketBanner);
+export default connected(AbuseTicketBanner) as React.ComponentType<{}>;
```
Comment (martinmckenna) [inline packages/manager/src/components/AbuseTicketBanner/AbuseTicketBanner.tsx]: upgrading TypeScript caught a couple issues.

File: .gitignore [hunk 2]
```diff
@@ -21,6 +21,7 @@ restart.sh
 .env.development.local
 .env.test.local
 .env.production.local
+*env*
```
[the remark this answers is not in the document]
Comment (martinmckenna) [reply .gitignore]: true. Maybe:

```
.env*
*env.
```

File: packages/manager/src/components/AbuseTicketBanner/AbuseTicketBanner.tsx (hunk 1 above)
Comment (acourdavault) [reply packages/manager/src/components/AbuseTicketBanner/AbuseTicketBanner.tsx]: @martinmckenna 
Here i had the issue with abuseTickets that had to be set. Why this cast and not making abuseTicket in props optional with `abuseTickets?:Notifications[]` ?

PS I m sure you have a good reason, i am just trying to understand why casting

Comment (martinmckenna) [reply packages/manager/src/components/AbuseTicketBanner/AbuseTicketBanner.tsx]: `abuseTickets` is not a prop that needs to be passed to this component directly. AbuseTicketBanner is getting `abuseTickets` by subscribing to Redux state. In other words, it's doing the work to get that prop internally.

The `connected` function is a higher-order function that returns the component with the `abuseTicket` prop attached.

**Pull request merged**